### PR TITLE
REFACTOR-#6889: Define `__all__` in `modin.config.__init__.py`

### LINKED ISSUE #6889: Define `__all__` for `envvars.py`

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

Just to follow good practice.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6889 
- [x] tests ~added and~ passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/config/envvars.py [hunk 1]
```diff
@@ -863,3 +863,53 @@ def _check_vars() -> None:
 
 
 _check_vars()
+
+
+__all__ = [
```
Comment (YarShev) [inline modin/config/envvars.py:868]: Should we move this to the init file?

File: modin/core/execution/ray/common/utils.py [hunk 2]
```diff
@@ -35,8 +35,8 @@
     RayRedisAddress,
     RayRedisPassword,
     StorageFormat,
-    ValueSource,
 )
+from modin.config.pubsub import ValueSource
```
Comment (YarShev) [inline modin/core/execution/ray/common/utils.py:39]: Why should we not include this into "all"?

**Comment by YarShev** [comment]
Why do we pin pytest in this PR? This is not related to the issue to resolve.

**Comment by anmyachev** [comment]
> Why do we pin pytest in this PR? This is not related to the issue to resolve.

Sure, it is needed to verify my last changes. I marked the commit with `REVERT ME` comment :)

File: modin/config/__init__.py [hunk 3]
```diff
@@ -13,5 +13,95 @@
...
+    RayRedisPassword,
+    ReadSqlEngine,
+    StorageFormat,
+    TestDatasetSize,
+    TestRayClient,
+    TestReadFromPostgres,
+    TestReadFromSqlServer,
+    TrackFileLeaks,
+)
+from modin.config.pubsub import Parameter, ValueSource  # noqa: F401
```
Comment (YarShev) [inline modin/config/__init__.py:56]: Why noqa?

File: modin/config/envvars.py [hunk 4]
```diff
@@ -23,7 +23,7 @@
 from packaging import version
 from pandas.util._decorators import doc  # type: ignore[attr-defined]
 
-from .pubsub import (
+from modin.config.pubsub import (
```
Comment (YarShev) [inline modin/config/envvars.py:26]:
```suggestion
from modin.config import (
```
Does this work?

File: modin/config/pubsub.py [hunk 5]
```diff
@@ -28,7 +28,7 @@
 )
 
 if TYPE_CHECKING:
-    from .envvars import EnvironmentVariable
+    from modin.config.envvars import EnvironmentVariable
```
Comment (YarShev) [inline modin/config/pubsub.py:31]:
```suggestion
    from modin.config import EnvironmentVariable
```
?

File: modin/config/__init__.py (hunk 3 above)
Comment (anmyachev) [reply modin/config/__init__.py]: Removed

File: modin/config/envvars.py (hunk 4 above)
Comment (anmyachev) [reply modin/config/envvars.py:26]: Such imports will lead to cyclical imports.

File: modin/config/pubsub.py (hunk 5 above)
Comment (anmyachev) [reply modin/config/pubsub.py:31]: Such imports will lead to cyclical imports.

**Review by YarShev — APPROVED** [review]

**Pull request merged**